server/utils/urlScraping.js

```diff
@@ -11,27 +12,45 @@ const scrapingAmazon = async (url) => {
   // Go to Amazon Product page
   try {
     await page.goto(url, {
-      waitUntil: ["load", "domcontentloaded", "networkidle0", "networkidle2"]
+      waitUntil: ["load", "domcontentloaded", "networkidle0", "networkidle2"],
     });
     await page.waitForSelector("#productTitle");
     await page.waitForSelector("#priceblock_ourprice");
 
     // Extract information from page
     const { title, price, imageUrl } = await page.evaluate(() => {
+      let price;
       const title = document.getElementById("productTitle").innerText;
-      const price = document.getElementById("priceblock_ourprice").innerText;
+
+      if (document.getElementById("priceblock_ourprice") != null) {
+        price = document.getElementById("priceblock_ourprice").innerText;
+      } else if (document.getElementById("priceblock_saleprice" != null)) {
+        price = document.getElementById("priceblock_saleprice").innerText;
+      } else {
+        price = "Product Not Available";
+      }
+
       const imageUrl = document
         .getElementById("landingImage")
         .getAttribute("src");
+
       return { title, price, imageUrl };
     });
-    await browser.close();
-    return { title, price, imageUrl };
   } catch (err) {
     console.error(err);
-    await browser.close();
     return null;
   }
+
+  // Create object for return
+  const product = new Product({
+    title,
+    price,
+    imageUrl,
+  });
+
+  await browser.close();
+
+  return product;
```

Comment:
you don't have to create a Product object from the product model. you can just return a simple object and the rest will be taken care of in the taskQueue.js where the actual Product document will get updated in our database. so I think the return should just be:
```return { title, price, imageUrl };```

Reply:
Good to know, I didn't know how it was going to be handled, I am changing this return.

Reply:
Make sure you pull from main because I've made changes to the return statement in urlScraping to match what I needed in taskQueue.js and the scraping API calls